src/main/scala/org/broadinstitute/hail/driver/GroupTestSKATO.scala

```diff
@@ -0,0 +1,359 @@
+    }
+
+    val symTab = Map(
+      "s" ->(0, TSample),
+      "sa" ->(1, vds.saSignature),
+      "va" -> (2, vds.vaSignature))
+
+    val ec = EvalContext(symTab)
+    val a = ec.a
+
+    val (yT, yQ) = Parser.parse(options.ySA, ec)
+
+    val yToDouble = toDouble(yT, options.ySA)
+    val ySA = vds.sampleIdsAndAnnotations.map { case (s, sa) =>
+      a(0) = s
+      a(1) = sa
+      yQ().map(yToDouble)
+    }
+
+    val (covT, covQ) = Parser.parseExprs(options.covSA, ec).unzip
+    val covToDouble = (covT, options.covSA.split(",").map(_.trim)).zipped.map(toDouble)
+    val covSA = vds.sampleIdsAndAnnotations.map { case (s, sa) =>
+      a(0) = s
+      a(1) = sa
+      (covQ.map(_ ()), covToDouble).zipped.map(_.map(_))
+    }
+    val nCovar = covT.size
+
+    val (yDefined, covDefined, samplesDefined) =
+      (ySA, covSA, sampleIds)
+        .zipped
+        .filter((y, c, s) => y.isDefined && c.forall(_.isDefined))
+
+    if (samplesDefined.length < 1) {
+      val sb = new StringBuilder()
+      sb.append(s"No samples with non-missing phenotype [${options.ySA}]")
+      if (nCovar > 0)
+        sb.append(s" and covariates [${options.covSA}]")
+      fatal(sb.result())
+    }
+    info(s"${samplesDefined.length} samples with non-missing phenotypes")
+
+    val seed = if (options.randomize) Random.nextInt() else options.seed
+    info(s"Using a random seed of [$seed]")
+
+    val cmd = Array(R, skatoScript)
+
+    val localBlockSize = options.blockSize
+
+    val ySABC = state.sc.broadcast(ySA)
+    val covSABC = state.sc.broadcast(covSA)
+    val splatBC = state.sc.broadcast(options.splat)
+
+    def printContext(w: (String) => Unit) = {
+      val y = pretty(JArray(ySABC.value.map { y => if (y.isDefined) JDouble(y.get) else JNull }.toList))
+      val cov = pretty(JArray(covSABC.value.map { cov => JArray(cov.map { c => if (c.isDefined) JDouble(c.get) else JNull }.toList) }.toList))
+      val covString = if (nCovar > 0) s""""COV":$cov,""" else ""
+      w(
+        s"""{"yType":"$yType",
+            |"nResampling":$nResampling,
+            |"typeResampling":"$typeResampling",
+            |"adjustment":$adjustment,
+            |"kernel":"$kernel",
+            |"method":"$method",
+            |"weightsBeta":[$weightsBeta],
+            |"imputeMethod":"$imputeMethod",
+            |"rCorr":[$rCorr],
+            |"missingCutoff":$missingCutoff,
+            |"estimateMAF":$estimateMAF,
+            |"seed":$seed,
+            |"Y":$y,
+            |$covString
+            |"groups":{""".stripMargin)
+    }
+
+    def printSep(w: (String) => Unit) = {
+      w(",")
+    }
+
+    def printFooter(w: (String) => Unit) = {
+      w("}}")
+    }
+
+    def printElement(w: (String) => Unit, g: (Any, Iterable[Iterable[Int]])) = {
+      val a = JArray(g._2.map(a => JArray(a.map(JInt(_)).toList)).toList)
+      w( s""""${g._1}":${pretty(a)}""")
+    }
+
+    val (baseType, querier) = vds.queryVA(options.groupKey)
```

Comment:
I'd rename `querier` something more evocative since you have a number of queriers in scope.  How about `qGroupKey`?